metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_legacy.py

```diff
@@ -228,36 +228,32 @@ def _populate_table_lineage(self):
     def get_table_upstream_workunits(self, discovered_tables):
         if self.config.include_table_lineage:
             for dataset_name in discovered_tables:
-                if self._is_dataset_pattern_allowed(
```

Comment:
This pattern check is not required, as its already present on discovered_tables [here](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py#L529).